render/src/shape_utils.rs

```diff
@@ -1006,16 +1006,54 @@ fn winding_number_line(
     // A downward segment (+y) decrements the winding number (including the final endpoint)
     // Perp-dot indicates which side of the segment the point is on.
     if begin.y < test_point.y {
-        if end.y >= test_point.y && d1.dx.get() * d0.dy.get() > d1.dy.get() * d0.dx.get() {
+        if end.y >= test_point.y
+            && (d1.dx.get() as i64) * (d0.dy.get() as i64)
+                > (d1.dy.get() as i64) * (d0.dx.get() as i64)
+        {
             return 1;
         }
-    } else if end.y < test_point.y && d1.dx.get() * d0.dy.get() < d1.dy.get() * d0.dx.get() {
+    } else if end.y < test_point.y
+        && (d1.dx.get() as i64) * (d0.dy.get() as i64) < (d1.dy.get() as i64) * (d0.dx.get() as i64)
+    {
         return -1;
     }
 
     0
 }
 
+#[cfg(test)]
+mod test {
```

Comment:
Can this be placed at bottom of the file please?

Reply:
I initially placed it at the bottom, but then I happened upon other tests in the same file that were placed further up:

https://github.com/ruffle-rs/ruffle/blob/ba3aa1b73b6ebcbe80f72ce6afa8ed680611667a/render/src/shape_utils.rs#L535-L571

Should I still move it down to the bottom? And should I leave the existing tests where they are (to keep the PR clean)?

Reply:
Oh, I didn't notice that. So let's keep this as-is. However I did just notice that the vast majority of codebase uses `mod tests` and not `mod test` (without an `s`). Can you please add an `s` to align to the rest of the codebase?

Reply:
I think I will have to move them all down:

```
error[E0428]: the name `tests` is defined multiple times
    --> render/src/shape_utils.rs:1020:1
     |
536  | mod tests {
     | --------- previous definition of the module `tests` here
...
1020 | mod tests {
     | ^^^^^^^^^ `tests` redefined here
     |
     = note: `tests` must be defined only once in the type namespace of this module

For more information about this error, try `rustc --explain E0428`.
error: could not compile `ruffle_render` due to previous error
warning: build failed, waiting for other jobs to finish...
```

Reply:
I got the `mod test` from here: https://github.com/ruffle-rs/ruffle/blob/f969bdce710b7911e7fe6ad96a95ba11fa67acb2/render/src/bitmap.rs#L501 , I did not realize the difference.

Reply:
I have moved all the tests down to the button, and ensured that the name is `mod tests`.